Fix more kwargs issues: git worktree and authorizer

kbrock opened the pull request:
```
lib/git_worktree.rb:323: warning: Using the last argument as keyword parameters is deprecated;
maybe ** should be added to the call

lib/rbac/authorizer.rb:43: warning: Using the last argument as keyword parameters is deprecated;
maybe ** should be added to the call
```

sorry I introduced these fixes in more than one PR. I am just putting in the fixes when I see the warnings show up.

Fryguy reviewed (approved):
LGTM, but would like another set of eyes, as I'm not sure what side-effect this might have - @NickLaMuro ?

NickLaMuro commented:
> sorry I introduced these in a few PRs. just putting them in when I see them.

@kbrock Can you add a link to where this was added by you?

NickLaMuro commented on lib/git_worktree.rb:
```diff
@@ -320,7 +320,7 @@ def fetch_and_merge
   def fetch
     with_remote_options do |remote_options|
-      @repo.fetch(@remote_name, remote_options)
+      @repo.fetch(@remote_name, **remote_options)
```
I don't think this one was you:  https://github.com/ManageIQ/manageiq/commit/bdb307e2c516ea549abbb99802163b6d67efac23

NickLaMuro reviewed (commented):
So I am going to need a little more time to test/review this as I am unsure the ramifications with `lib/git_worktree.rb`, as we don't own that code and it is a little confusing.

There is a particular part of the deprecation docs in Ruby that is of note here:

> https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/
>
> Note that Ruby 3.0 doesn’t behave differently when calling a method which doesn’t accept keyword arguments with keyword arguments. For instance, the following case is not going to be deprecated and will keep working in Ruby 3.0. The keyword arguments are still treated as a positional Hash argument.
> 
> ```ruby
> def foo(kwargs = {})
>   kwargs
> end
> 
> foo(k: 1) #=> {:k=>1}
> ```

Which is unclear what we are working with because `Rugged::Repository#fetch` ends up calling `Rugged::Remote#fetch`, which is defined in `C` here:

https://github.com/libgit2/rugged/blob/b92520bbbee0d1f88f2e7ffb42cceb32e993ce1b/ext/rugged/rugged_remote.c#L600-L613

So I want to test that this is okay, but probably is.  Will report back when I feel comfortable with this line change.

kbrock commented:
> > sorry I introduced these [fixes] in a few PRs. just putting them in when I see them.
> 
> @kbrock Can you add a link to where this was added by you?

@NickLaMuro I changed the description.

I was trying to say that I have seen a number of these warnings and put in a PR when I see them rather than collecting them all in one go. That also says that I am not proactively searching for them.

I have seen PRs like this across a number of gems (e.g.: ancestry, rails, active_hash) and quite a few in our code base.
While I agree that we want to ensure that this is the right, the suggestion here tends to be spot on. I had thought that rails did not run with ruby 3.0 until a few `**` fixes were introduced. But your link suggests that my memory is not quite right here.

This file has a few other methods that follow a similar coding pattern. Those others may need to be fixed as well but since I have not seen warnings, I am not making additional changes.

NickLaMuro reviewed (approved):
Finally got around to doing an isolated test with this, and threw it in a gist for reference:

https://gist.github.com/NickLaMuro/ca20303520942fcd2d2418a70716cae2

The gist repo above _**IS**_ the test repo in this case, and assumes a `~/.ssh/id_rsa` key file attached to github.  But worked for me and showed that it still accepts the options like that, so I think we are good with this.
